### ISSUE TITLE
Add type for returning arbitrary size tuples

### ISSUE THREAD

**Issue by mightybyte** [tathougies/beam#450]
The `postgresql-simple` library has a type that facilitates returning query results of arbitrary size.

https://hackage.haskell.org/package/postgresql-simple-0.6.2/docs/Database-PostgreSQL-Simple.html#t::.

Could something similar be added to beam?  I'm not stuck on `:.` as the operator.  Servant has a `:.` operator that conflicts, so I'm open to anything.  But the convenience would be really nice.

**Comment by kmicklas** [comment]
You should effectively be able to do this already by nesting tuples. @mightybyte Are there any benefits to a specific type for this that I'm missing?

**Comment by mightybyte** [comment]
Nesting tuples is just a pain.  Also, when I was playing with it I believe there was a mismatch between the size of tuples that were actually supported compared with what the documentation said.  An operator like this is purely for convenience.

**Comment by kmicklas** [comment]
Yeah, I guess having an operator with associativity is nice because you don't have to write out the nesting structure. PRs welcome!

**Comment by 3noch** [comment]
Is there a library that provides this type? It seems like something that doesn't need to be rebuilt.
